### ISSUE TITLE
Cargo valgrind crashed while running tests.

### ISSUE THREAD

**Issue by Superhepper** [jfrimmel/cargo-valgrind#55]
Thank you for an excellent tool!

As I was running this on our integration tests I experienced a crash with the following message:
```
cargo-valgrind: version 2.0.3
XML format mismatch between `valgrind` and `cargo valgrind`: custom: duplicate field `stack`
XML output of valgrind:
```xml
<?xml version="1.0"?>

<valgrindoutput>

<protocolversion>4</protocolversion>
<protocoltool>memcheck</protocoltool>

<preamble>
  <line>Memcheck, a memory error detector</line>
  <line>Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.</line>
  <line>Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info</line>
  <line>Command: /home/hepper/workspace/rust-tss-esapi/target/debug/deps/integration_tests-2d5723b6547f6cbb --test-threads=1 --nocapture</line>
</preamble>

<pid>9946</pid>
<ppid>9945</ppid>
<tool>memcheck</tool>

<args>
  <vargv>
    <exe>/usr/bin/valgrind.bin</exe>
    <arg>--xml=yes</arg>
    <arg>--xml-socket=127.0.0.1:41415</arg>
  </vargv>
  <argv>
    <exe>/home/hepper/workspace/rust-tss-esapi/target/debug/deps/integration_tests-2d5723b6547f6cbb</exe>
    <arg>--test-threads=1</arg>
    <arg>--nocapture</arg>
  </argv>
</args>

<status>
  <state>RUNNING</state>
  <time>00:00:00:01.897 </time>
</status>
```

**Comment by spital** [comment]
may be related:
https://github.com/jfrimmel/cargo-valgrind/issues/68#issuecomment-1164739929

**Comment by jfrimmel** [comment]
Could you try out the release 2.1.0? This nicht contain the fix for your issue.

**Comment by TommyAmberson** [comment]
I'm still getting this XML format mismatch error
```
$ cargo install --git https://github.com/jfrimmel/cargo-valgrind cargo-valgrind
    Updating git repository `https://github.com/jfrimmel/cargo-valgrind`
     Ignored package `cargo-valgrind v2.1.0 (https://github.com/jfrimmel/cargo-valgrind#43b31c93)` is already installed, use --force to override
```
What about the valgrind version?
```
$ valgrind --version
valgrind-3.18.1
```

**Comment by Superhepper** [comment]
Running with valgrind 3.15.0 and cargo-valgrind 2.1.0 actually made it worse. It crashes silently just after all the tests had been run and it was about to present the report.

**Comment by spital** [comment]
pls try at least valgrind 3.17, see https://github.com/jfrimmel/cargo-valgrind/issues/55#issuecomment-1164741044

**Comment by Superhepper** [comment]
just tried it with valgrind 3.19 and wiht no difference. I got no XML output.

**Comment by jfrimmel** [comment]
Hey, the error with "XML format mismatch between `valgrind` and `cargo valgrind`: custom: duplicate field `stack`
" should be fixed as of `cargo valgrind` v2.2.0. Consider upgrading and testing you code. I'll go ahead and close this issue, but feel free to re-open if there are ongoing issues.

Thanks for your patience.

**Comment by jfrimmel** [comment]
With regards to newer valgrind versions: I always try with the latest valgrind version available, which is currently 3.23:
```shell
$ valgrind --version
valgrind-3.23.0
```

If your issue persists with 2.2.0 or newer versions of `cargo valgrind`, please report a new issue with details, so that I can reproduce and fix it.

**Comment by Superhepper** [comment]
This was a long time ago. We fixed the problem causing the memory leak and we no longer had the gigantic reports this error went away.

And since then cargo valgrind have not had any problems in our CI pipeline that I know of. So I think it is safe to close this if no one else have encountered this.
